### PR TITLE
[fix](order by) fix bug of order by desc when rowsets is no overlapping

### PR DESCRIPTION
**Pull request by jacktengg**
# Proposed changes

Issue Number: close #xxx

## Problem summary
In the case of rowets non-overlap and desc sorting, the logic of VCollectIterator::Level0Iterator::init_for_union will be followed. In this function, the row ref pos of the first level0 iterator is set to 0, and the row pos of other level0 iterators are all Set to -1.

But in the level1iterator, when rowets are non-overlapping and is ordering by desc, the list of rowset iterators will be reversed, causing the row ref pos of the first level0 iterator in the list to be -1, causing the block reader to think that the entire tablet has no data.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yiguolei** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

File: regression-test/suites/query_p0/sort/sort.groovy [hunk 1]
```diff
@@ -67,4 +67,36 @@ suite("sort") {
...
+      `dev` varchar(255) NOT NULL,
+      `dec0` decimal(10, 3) REPLACE_IF_NOT_NULL NULL,
+      `dec1` decimal(10, 3) REPLACE_IF_NOT_NULL NULL,
+      `dec2` decimal(10, 3) REPLACE_IF_NOT_NULL NULL,
+      `dec3` decimal(10, 3) REPLACE_IF_NOT_NULL NULL,
+      `update_time` datetime REPLACE NULL
+    ) ENGINE=OLAP
+    AGGREGATE KEY(`time_period`, `area_name`, `province`, `res_name`, `dev`)
+    DISTRIBUTED BY HASH(`time_period`, `area_name`, `province`, `res_name`, `dev`) BUCKETS 4
+    PROPERTIES (
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sort/sort.groovy:86]: add disable compaction;

remove distributed by

**Review by yixiutt — DISMISSED** [review]

**Comment by yixiutt** [comment]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buidlall

**Pull request merged**